arduino/libraries/libraries_test.go

```diff
@@ -85,3 +88,89 @@ func TestLibrariesLoader(t *testing.T) {
 		require.True(t, lib.IsLegacy)
 	}
 }
+
+func TestSymlinkLoop(t *testing.T) {
+	// Set up directory structure of test library.
+	libraryPath := paths.New("testdata", "TestLib")
+	tempLibPath, err := paths.TempDir().MkTempDir("TestSymlinkLoop")
+	require.NoError(t, err)
+	defer tempLibPath.RemoveAll() // Clean up after the test.
+	testLib := tempLibPath.Join("TestLib")
+	require.NoError(t, libraryPath.CopyDirTo(testLib))
+	examplesPath := testLib.Join("examples")
+	require.NoError(t, examplesPath.Mkdir())
+
+	// It's probably most friendly for contributors using Windows to create the symlinks needed for the test on demand.
+	err = os.Symlink(examplesPath.Join("..").String(), examplesPath.Join("UpGoer1").String())
+	require.NoError(t, err, "This test must be run as administrator on Windows to have symlink creation privilege.")
+	// It's necessary to have multiple symlinks to a parent directory to create the loop.
+	err = os.Symlink(examplesPath.Join("..").String(), examplesPath.Join("UpGoer2").String())
+	require.NoError(t, err)
+
+	// The failure condition is Load() never returning, testing for which requires setting up a timeout.
+	done := make(chan bool)
+	go func() {
+		_, err = Load(testLib, User)
+		done <- true
+	}()
+
+	noHang := assert.Eventually(
+		t,
+		func() bool {
+			select {
+			case <-done:
+				return true
+			default:
+				return false
+			}
+		},
+		20*time.Second,
+		10*time.Millisecond,
+		"Infinite symlink loop while loading library",
+	)
+	require.True(t, noHang)
```

Comment:
IIUC this assert will fail after 20 seconds, that seems way too much, we can assume that `Load` should complete in a few milliseconds. I'd suggest to use 2 seconds instead.
Also the check may be done much more easily like this:
```suggestion
	select {
	case <-done:
	case <-time.After(2*time.Second):
		require.FailNow(t, "Load didn't complete in the allocated time.")
	}
```